lib/rust/profiler/src/lib.rs

```diff
@@ -246,16 +271,80 @@ pub mod js {
 // ===============
 
 /// Identifies an event in the profiling log.
-#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+///
+/// This is a packed representation of [`id::Event`], whose structure better represents the types of
+/// event reference.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash, serde::Serialize, serde::Deserialize)]
 pub struct EventId(u32);
 
+impl EventId {
+    /// Special value indicating that no explicit prior event is associated.
+    ///
+    /// When used to identify a parent, this indicates that the parent can be inferred to be the
+    /// current profiler.
+    pub const fn implicit() -> Self {
+        IMPLICIT_ID
+    }
+
+    /// Interpret this value.
+    pub fn into_event(self) -> id::Event {
+        if self == IMPLICIT_ID {
+            id::Event::Implicit
+        } else {
+            id::Event::Explicit(if self == APP_LIFETIME_ID {
+                id::Explicit::AppLifetime
+            } else {
+                id::Explicit::Runtime(id::Runtime(self.0))
+            })
+        }
+    }
+}
+
+
+// === mod id ===
+
+/// Facilities for interpreting an event ID.
```

Comment:
Could you extend this docs please? It's worth describing what kin of interpretation does it handle and provide to the reader a little more context.